### ISSUE TITLE
EF Core 7 generates UpdateData statement in every migration

### ISSUE THREAD

**Issue by mhoerandner** [dotnet/efcore#30217]
## EF Core 7 generates `UpdateData` statement in every migration

Since EF Core 7 an `migrationBuilder.UpdateData`-statement is generated in every migration for my seeded entity `Author`, due to a property-name with a special character (e.g. `_LegacyName`). 

I recently updated from EF Core 6 to EF Core 7. The problem did not exist in previous EF Core versions either (5.0, 2.x) .

### Example

```C#
public class Author
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public string? _LegacyName { get; set; }
}
```

```C#
public class AuthorDbContext : DbContext
{
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);

        modelBuilder.Entity<Author>(e =>
        {
            e.HasData(new[]
            {
                new Author { Id = 1, Name = "Name", _LegacyName = "LegacyName" }
            });
        });
    }

    // ...
}
```

### Result

EF Core tries to update the seed-data in every migration, although the seed data has not changed.

```C#
public partial class Migration1 : Migration
{
        /// <inheritdoc />
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.UpdateData(
                table: "Author",
                keyColumn: "Id",
                keyValue: 1,
                column: "_LegacyName",
                value: "LegacyName");
        }

        // ...
}
```

#### Modelsnapshot

The property-name is spelled wrong in the call of `HasData` in the modelsnapshot.

![grafik](https://user-images.githubusercontent.com/44393602/216963079-5e135242-accc-484b-b545-22a156e88729.png)

### Provider and version information

EF Core version: 7.0.2
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 7.0

**Comment by mhoerandner** [comment]
The full example here:
[EfCorePropertyIssue.zip](https://github.com/dotnet/efcore/files/10662739/EfCorePropertyIssue.zip)

**Comment by ajcvickers** [comment]
Duplicate of #29450. Confirmed fixed with local build of 7.0.3.
